### ISSUE TITLE
InternalError: a fault occurred in a recent unsafe memory access operation in compiled Java code

### ISSUE THREAD

**Issue by tivervac** [ehcache/ehcache3#2948]
We use EhCache 3.8.1 in our code and we're encountering the following stacktrace. It doesn't seem like something our product could influence, we don't call any unsafe code.

Since this is a stacktrace from a client, we do not know how to reproduce this issue.

```
java.lang.Throwable: Caused by: java.lang.InternalError: a fault occurred in a recent unsafe memory access operation in compiled Java code
    at java.base/java.nio.DirectIntBufferS.get(java/base/java/nio/DirectIntBufferS.java:285)
    at org.ehcache.shadow.org.terracotta.offheapstore.OffHeapHashMap.readLong(org/ehcache/shadow/org/terracotta/offheapstore/OffHeapHashMap.java:993)
    at org.ehcache.shadow.org.terracotta.offheapstore.OffHeapHashMap.computeIfPresentWithMetadata(org/ehcache/shadow/org/terracotta/offheapstore/OffHeapHashMap.java:2101)
    at org.ehcache.shadow.org.terracotta.offheapstore.AbstractLockedOffHeapHashMap.computeIfPresentWithMetadata(org/ehcache/shadow/org/terracotta/offheapstore/AbstractLockedOffHeapHashMap.java:542)
    at org.ehcache.shadow.org.terracotta.offheapstore.concurrent.AbstractConcurrentOffHeapMap.computeIfPresentWithMetadata(org/ehcache/shadow/org/terracotta/offheapstore/concurrent/AbstractConcurrentOffHeapMap.java:781)
    at org.ehcache.impl.internal.store.disk.EhcachePersistentConcurrentOffHeapClockCache.computeIfPresentAndPin(org/ehcache/impl/internal/store/disk/EhcachePersistentConcurrentOffHeapClockCache.java:192)
    at org.ehcache.impl.internal.store.offheap.AbstractOffHeapStore.getAndFault(org/ehcache/impl/internal/store/offheap/AbstractOffHeapStore.java:826)
    at org.ehcache.impl.internal.store.tiering.TieredStore.lambda$get$0(org/ehcache/impl/internal/store/tiering/TieredStore.java:89)
    at org.ehcache.impl.internal.store.heap.OnHeapStore.lambda$getOrComputeIfAbsent$12(org/ehcache/impl/internal/store/heap/OnHeapStore.java:703)
    at org.ehcache.impl.internal.store.heap.OnHeapStore$Fault.getValueHolder(org/ehcache/impl/internal/store/heap/OnHeapStore$Fault.java:1017)
```

**Comment by chrisdennis** [comment]
There is a bug in the version of offheap-store used by 3.8.1 that has been fixed in more recent version of Ehcache: Terracotta-OSS/offheap-store#52

As a first step I would suggest updating to a version from the 3.9 line to see if that clears up the problem.

**Comment by tivervac** [comment]
Thanks for the quick response. I'll try that though our upgrade is a bit troubled by https://github.com/ehcache/ehcache3/issues/2938

**Comment by chrisdennis** [comment]
Closing under the assumption that this is fixed in 3.9 (and beyond). Please reopen if this issue still persists.
